test/xchar-test.cc

```diff
@@ -102,11 +102,62 @@ struct custom_char {
   template <typename T>
   constexpr custom_char(T val) : value(static_cast<int>(val)) {}
 
-  operator char() const {
+  constexpr operator char() const {
     return value <= 0xff ? static_cast<char>(value) : '\0';
   }
+  constexpr bool operator<(custom_char c) const { return value < c.value; }
 };
 
+#  ifdef _LIBCPP_VERSION
+namespace std {
+
+template <> struct char_traits<custom_char> {
```

Comment:
Does the standard permit user specializations of `char_traits`?

Reply:
https://en.cppreference.com/w/cpp/string/char_traits:

> The char_traits class template serves as a basis for explicit instantiations. The user can [provide a specialization](https://en.cppreference.com/w/cpp/language/extending_std) for any custom character types. Several explicit specializations are provided for the standard character types (see below), other specializations are not required to satisfy the requirements of [CharTraits](https://en.cppreference.com/w/cpp/named_req/CharTraits).